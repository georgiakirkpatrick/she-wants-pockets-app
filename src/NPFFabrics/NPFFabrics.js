import React from 'react'
import FormButton from '../FormButton/FormButton'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormDropdown from '../FormDropdown/FormDropdown'
import FormFieldset from '../FormFieldset/FormFieldset'
import FormPage from '../FormPage/FormPage'
import FormPopUp from '../FormPopUp/FormPopUp'
import NPFFooter from '../NPFFooter/NPFFooter'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import FormTextInput from '../FormTextInput/FormTextInput'
import FormUrlInput from '../FormUrlInput/FormUrlInput'
import formData from '../FORM_DATA'

const NPFFabrics = props => {
    const fabChange = event => {
        const fabFields = {...props.fabFact}
        fabFields[event.target.name] = event.target.value
        props.setFabFact(fabFields)
    }

    const certChange = event => {
        props.setCertChecks({...props.certChecks, [event.target.id]: !props.certChecks[event.target.id]})
    }

    const handleClose = () => {
        props.setCertPopUp(false)
        props.setFactPopUp(false)
        props.setMillPopUp(false)
    }

    const factPopUpStatus = () => {
        if (props.factPopUp === true) {
            return 'FormPopUp__pop-up active'
        }
        return 'FormPopUp__pop-up'
    }

    const newFactChange                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      = event => {
        const newFields = {...props.newFact}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      
        newFields[event.target.name] = event.target.value
        props.setNewFact(newFields)
    }

    const factPopUp = (id, status, change) => (
        <FormPopUp
            id={id + 'NewFact'} 
            status={status}
            title='New Factory'
            click={() => handleClose()}
            buttonText='Submit Factory'
        >
            <FormPromptWithSub
                prompt='Add a factory'
                promptSubtitle=''
            />
            <FormTextInput 
                id={id + 'NewFactName'}
                name='factoryName'
                prompt='Name'
                currentValue={props.newFact.factoryName}
                handleChange={event => change(event)} 
            />
            <FormDropdown
                id={id + 'NewFactLocation'}
                name='factoryLocation'
                prompt='Location'
                currentValue={props.newFact.factoryLocation}
                options={props.countries}
                handleChange={event => change(event)} 
            />
            <FormUrlInput
                id={id + 'NewFactWebsite'}
                name='factoryWebsite'
                prompt='Website'
                currentValue={props.newFact.factoryWebsite}
                handleChange={event => change(event)}
            />
            <FormTextInput 
                id={id + 'NewFactNotes'}
                name='factoryNotes'
                prompt='Notes'
                currentValue={props.newFact.factoryNotes}
                handleChange={event => change(event)} 
            />
        </FormPopUp>
    )

    const millPopUpStatus = () => {
        if (props.millPopUp === true) {
            return 'FormPopUp__pop-up active'
        }
        return 'FormPopUp__pop-up'
    }

    const newMillChange = event => {
        const newFields = {...props.newMill}
        newFields[event.target.name] = event.target.value
        props.setNewMill(newFields)
    }

    const millPopUp = (id, status, change) => (
        <FormPopUp
            id={id + 'NewMill'} 
            status={status}
            title='New Fabric Mill'
            click={() => handleClose()}
            buttonText='SUBMIT FABRIC MILL'
        >
            <FormPromptWithSub
                prompt='Add a fabric mill'
                promptSubtitle=''
            />
            <FormTextInput 
                id={id + 'NewMillName'}
                name='factoryName'
                prompt='Name'
                currentValue={props.newFact.factoryName}
                handleChange={event => change(event)} 
            />
            <FormDropdown
                id={id + 'NewMillLocation'}
                name='factoryLocation'
                prompt='Location'
                currentValue={props.newFact.factoryLocation}
                options={props.countries}
                handleChange={event => change(event)} 
            />
            <FormUrlInput
                id={id + 'NewMillWebsite'}
                name='factoryWebsite'
                prompt='Website'
                currentValue={props.newFact.factoryWebsite}
                handleChange={event => change(event)}
            />
            <FormTextInput 
                id={id + 'NewMillNotes'}
                name='factoryNotes'
                prompt='Notes'
                currentValue={props.newFact.factoryNotes}
                handleChange={event => change(event)} 
            />
        </FormPopUp>
    )

    const certPopUpStatus = () => {
        if (props.certPopUp === true) {
            return 'FormPopUp__pop-up active'
        }
        return 'FormPopUp__pop-up'
    }

    const newCertChange = event => {
        const newCertFields = {...props.newCert}
        newCertFields[event.target.name] = event.target.value
        props.setNewCert(newCertFields)
    }

    const certPopUp = (id, status, change) => (
        <FormPopUp
            id={id + 'NewCert'} 
            status={status}
            title='New Certification'
            click={() => handleClose()}
            buttonText='SUBMIT CERTIFICATION'
        >
            <FormPromptWithSub
                prompt='Add a certification'
                promptSubtitle=''
            />
            <FormTextInput 
                id={id + 'NewCertName'}
                name='newCertName'
                prompt='Name'
                currentValue={props.newCert.newCertName}
                handleChange={event => change(event)} 
            />
            <FormUrlInput
                id={id + 'NewCertWebsite'}
                name='newCertWebsite'
                prompt='Website'
                currentValue={props.newCert.newCertWebsite}
                handleChange={event => change(event)}
            />
        </FormPopUp>
    )

    return (
        <div id='fabrics'>
            <FormPage title={props.title}>
                <FormPromptWithSub 
                    prompt='Please enter the fabric details provided on the product webpage.'
                    promptSubtitle='If the product page does not provide the information detailed below, please select "not disclosed" from the dropdown.'
                />
                <FormFieldset>
                    <FormDropdown 
                        id='dyeFinLocation'
                        name='dyeFinLocation'
                        prompt='Dyeing and finishing location'
                        currentValue={props.fabFact.dyeFinLocation}
                        options={props.countries}
                        handleChange={event => fabChange(event)} 
                    />

                    <FormDropdown 
                        id='dyeFinFactory'
                        name='dyeFinFactory'
                        prompt='Dyeing and finishing factory'
                        currentValue={props.fabFact.dyeFinFactory}
                        options={props.factories}
                        handleChange={event => fabChange(event)} 
                    />

                    <FormButton 
                        buttonText='ADD A FACTORY' 
                        click={() => props.setFactPopUp(true)}
                    />

                    {factPopUp(props.id, factPopUpStatus(), newFactChange)}
                
                    <FormDropdown 
                        id='wovKnitLocation'
                        name='wovKnitLocation'
                        prompt='Knitting or weaving location'
                        currentValue={props.fabFact.wovKnitLocation}
                        options={props.countries}
                        handleChange={event => fabChange(event)} 
                    />

                    <FormDropdown 
                        id='wovKnitFactory'
                        name='wovKnitFactory'
                        prompt='Fabric mill (knitting or weaving)'
                        currentValue={props.fabFact.wovKnitFactory}
                        options={props.factories}
                        handleChange={event => fabChange(event)} 
                    />

                    <FormButton 
                        buttonText='ADD A FABRIC MILL' 
                        click={() => props.setMillPopUp(true)}
                    />

                    {millPopUp(props.id, millPopUpStatus(), newMillChange)}

                    <FormPromptWithSub 
                        prompt='Does the primary fabric have any of the following certifications?'
                        promptSubtitle=''
                    />

                    <FormCheckboxSection
                        options={formData.manufacturing.certifications.options} 
                        selectedOptions={props.certChecks}
                        handleChange={event => certChange(event)}
                    />

                    <FormButton
                        buttonText='ADD A CERTIFICATION'
                        click={() => props.setCertPopUp(true)}
                    />

                    {certPopUp(props.id, certPopUpStatus(), newCertChange)}

                </FormFieldset>
            </FormPage>
            <NPFFooter buttons='prevNext' previousButton={() => props.setPage(props.currentPage - 1)} nextButton={() => props.setPage(props.currentPage + 1)} />
        </div>
    )    
}

NPFFabrics.defaultProps = {
    fabFact: {
        dyeFinLocation: '',
        dyeFinFactory: '',
        dyeFinNotes: '',
        wovKnitLocation: '',
        wovKnitFactory: '',
        wovKnitNotes: ''
    },
    setFabFact: () => {},
    factPopUp: false,
    setFactPopUp: () => {},
    newFact: {
        factoryName: '',
        factoryLocation: '',
        factoryWebsite: '',
        factoryNotes: ''
    },
    setNewFact: () => {},
    certChecks: {},
    setCertChecks: () => {},
    certPopUp: false,
    setCertPopUp: () => {},
    newCert: {
        newCertName: '',
        newCertWebsite: ''
    },
    setNewCert: () => {},
    millPopUp: false,
    setMillPopUp: () => {},
    newMill: {
        millName: '',
        millLocation: '',
        millWebsite: '',
        millNotes: ''
    },
    setNewMill: () => {},
}

export default NPFFabrics