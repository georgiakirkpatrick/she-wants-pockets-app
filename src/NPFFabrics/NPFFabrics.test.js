import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import NPFFabrics from './NPFFabrics'

describe('NPFFabrics', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<NPFFabrics />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it ('render the UI as expected', () => {
        const tree = renderer
            .create(<NPFFabrics />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})