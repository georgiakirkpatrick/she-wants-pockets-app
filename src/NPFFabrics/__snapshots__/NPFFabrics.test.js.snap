// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFFabrics render the UI as expected 1`] = `
<div
  id="fabrics"
>
  <section
    className="FormPage"
  >
    <div
      className="empty"
    />
    <form>
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Please enter the fabric details provided on the product webpage.
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          If the product page does not provide the information detailed below, please select "not disclosed" from the dropdown.
        </p>
      </div>
      <fieldset
        className="FormFieldset"
      >
        <div />
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="dyeFinLocation"
          >
            Dyeing and finishing location
          </label>
          <select
            name="dyeFinLocation"
            onChange={[Function]}
            value=""
          />
        </div>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="dyeFinFactory"
          >
            Dyeing and finishing factory
          </label>
          <select
            name="dyeFinFactory"
            onChange={[Function]}
            value=""
          />
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          ADD A FACTORY
        </button>
        <section
          className="FormPopUp__pop-up"
          id="undefinedNewFact"
        >
          <button
            className="FormPopUp__close-button"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-times fa-w-11 fa-lg "
              data-icon="times"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 352 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
          <h1>
            New Factory
          </h1>
          <div
            className="FormPromptWithSub"
          >
            <p
              className="FormPromptWithSub__prompt"
            >
              Add a factory
            </p>
            <p
              className="FormPromptWithSub__subtitle"
            >
              
            </p>
          </div>
          <div
            className="FormText"
          >
            <label
              htmlFor="undefinedNewFactName"
            >
              Name
            </label>
            <input
              defaultValue=""
              id="undefinedNewFactName"
              name="factoryName"
              onChange={[Function]}
              type="text"
            />
          </div>
          <div
            className="FormDropdown"
          >
            <label
              htmlFor="undefinedNewFactLocation"
            >
              Location
            </label>
            <select
              name="factoryLocation"
              onChange={[Function]}
              value=""
            />
          </div>
          <div
            className="FormUrlInput"
          >
            <label
              htmlFor="undefinedNewFactWebsite"
            >
              Website
            </label>
            <input
              defaultValue=""
              id="undefinedNewFactWebsite"
              name="factoryWebsite"
              onChange={[Function]}
              type="url"
            />
          </div>
          <div
            className="FormText"
          >
            <label
              htmlFor="undefinedNewFactNotes"
            >
              Notes
            </label>
            <input
              defaultValue=""
              id="undefinedNewFactNotes"
              name="factoryNotes"
              onChange={[Function]}
              type="text"
            />
          </div>
          <button
            className="FormButton"
            onClick={[Function]}
            type="button"
          >
            Submit Factory
          </button>
        </section>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="wovKnitLocation"
          >
            Knitting or weaving location
          </label>
          <select
            name="wovKnitLocation"
            onChange={[Function]}
            value=""
          />
        </div>
        <div
          className="FormDropdown"
        >
          <label
            htmlFor="wovKnitFactory"
          >
            Fabric mill (knitting or weaving)
          </label>
          <select
            name="wovKnitFactory"
            onChange={[Function]}
            value=""
          />
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          ADD A FABRIC MILL
        </button>
        <section
          className="FormPopUp__pop-up"
          id="undefinedNewMill"
        >
          <button
            className="FormPopUp__close-button"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-times fa-w-11 fa-lg "
              data-icon="times"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 352 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
          <h1>
            New Fabric Mill
          </h1>
          <div
            className="FormPromptWithSub"
          >
            <p
              className="FormPromptWithSub__prompt"
            >
              Add a fabric mill
            </p>
            <p
              className="FormPromptWithSub__subtitle"
            >
              
            </p>
          </div>
          <div
            className="FormText"
          >
            <label
              htmlFor="undefinedNewMillName"
            >
              Name
            </label>
            <input
              defaultValue=""
              id="undefinedNewMillName"
              name="factoryName"
              onChange={[Function]}
              type="text"
            />
          </div>
          <div
            className="FormDropdown"
          >
            <label
              htmlFor="undefinedNewMillLocation"
            >
              Location
            </label>
            <select
              name="factoryLocation"
              onChange={[Function]}
              value=""
            />
          </div>
          <div
            className="FormUrlInput"
          >
            <label
              htmlFor="undefinedNewMillWebsite"
            >
              Website
            </label>
            <input
              defaultValue=""
              id="undefinedNewMillWebsite"
              name="factoryWebsite"
              onChange={[Function]}
              type="url"
            />
          </div>
          <div
            className="FormText"
          >
            <label
              htmlFor="undefinedNewMillNotes"
            >
              Notes
            </label>
            <input
              defaultValue=""
              id="undefinedNewMillNotes"
              name="factoryNotes"
              onChange={[Function]}
              type="text"
            />
          </div>
          <button
            className="FormButton"
            onClick={[Function]}
            type="button"
          >
            SUBMIT FABRIC MILL
          </button>
        </section>
        <div
          className="FormPromptWithSub"
        >
          <p
            className="FormPromptWithSub__prompt"
          >
            Does the primary fabric have any of the following certifications?
          </p>
          <p
            className="FormPromptWithSub__subtitle"
          >
            
          </p>
        </div>
        <div
          className="FormCheckboxSection"
        >
          <div
            className="empty"
          />
          <div
            className="FormCheckboxSection__options-container"
          >
            <div
              className="FormCheckboxSection__option"
            >
              <input
                id={1}
                name={1}
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor={1}
              >
                BlueSign
              </label>
            </div>
            <div
              className="FormCheckboxSection__option"
            >
              <input
                id={2}
                name={2}
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor={2}
              >
                OEKO-TEX
              </label>
            </div>
          </div>
        </div>
        <button
          className="FormButton"
          onClick={[Function]}
          type="button"
        >
          ADD A CERTIFICATION
        </button>
        <section
          className="FormPopUp__pop-up"
          id="undefinedNewCert"
        >
          <button
            className="FormPopUp__close-button"
            onClick={[Function]}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-times fa-w-11 fa-lg "
              data-icon="times"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 352 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </button>
          <h1>
            New Certification
          </h1>
          <div
            className="FormPromptWithSub"
          >
            <p
              className="FormPromptWithSub__prompt"
            >
              Add a certification
            </p>
            <p
              className="FormPromptWithSub__subtitle"
            >
              
            </p>
          </div>
          <div
            className="FormText"
          >
            <label
              htmlFor="undefinedNewCertName"
            >
              Name
            </label>
            <input
              defaultValue=""
              id="undefinedNewCertName"
              name="newCertName"
              onChange={[Function]}
              type="text"
            />
          </div>
          <div
            className="FormUrlInput"
          >
            <label
              htmlFor="undefinedNewCertWebsite"
            >
              Website
            </label>
            <input
              defaultValue=""
              id="undefinedNewCertWebsite"
              name="newCertWebsite"
              onChange={[Function]}
              type="url"
            />
          </div>
          <button
            className="FormButton"
            onClick={[Function]}
            type="button"
          >
            SUBMIT CERTIFICATION
          </button>
        </section>
      </fieldset>
    </form>
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter_previous"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-left fa-w-14 NPFFooter_arrow"
        data-icon="arrow-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter_next"
      onClick={[Function]}
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-arrow-right fa-w-14 NPFFooter_arrow"
        data-icon="arrow-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
</div>
`;
